src/selectors/getContextsSortedAndRanked.ts

```diff
@@ -1,12 +1,12 @@
 import { makeCompareByProp, sort } from '../util'
-import { getContexts } from '../selectors'
+import { getContexts, isContextVisibleForUser } from '../selectors'
 import { State } from '../util/initialState'
 import { ThoughtContext } from '../types'
 
 /** Gets all contexts that the given thought is in, sorted and ranked. */
 const getContextsSortedAndRanked = (state: State, value: string): ThoughtContext[] =>
   sort(
-    getContexts(state, value),
+    getContexts(state, value).filter(({ context }) => isContextVisibleForUser(state, context.concat(value))),
```

Comment:
When concatenating `Contexts` or `Paths` you always have to `unroot` the result since the ROOT token is only used in the ROOT context itself, not any of its descendants.